### PR TITLE
CLDC-3735: Sales - Move Q19 (What type of unit is the property?) (25/26)

### PR DESCRIPTION
**Pull request by Dinssa**
This branch moves the question _What type of unit is the property?_ to before _How many bedrooms does the property have?_. If a service user answers "bedsit" then we'll infer the number of bedrooms is 1 and skip the question.

As its no longer possible for someone to incorrectly choose type as bedsit and number of bedrooms as 1 I've also removed the hint message "A bedsit has 1 bedroom." from the number of bedrooms question.

### CODE REVIEW

File: spec/models/form/sales/subsections/property_information_spec.rb [hunk 1]
```diff
@@ -48,6 +49,7 @@
 
       before do
         allow(form).to receive(:start_year_2024_or_later?).and_return(true)
+        allow(form).to receive(:start_year_2025_or_later?).and_return(false)
```
Comment (RachaelBooth) [inline spec/models/form/sales/subsections/property_information_spec.rb:52]: Does this need a 2025 version of the test that changes?

Comment (Dinssa) [reply spec/models/form/sales/subsections/property_information_spec.rb:52]: Ye, added one

File: app/models/sales_log.rb [hunk 2]
```diff
@@ -396,6 +396,10 @@ def is_bedsit?
     proptype == 2
   end
 
+  def is_beds_inferred?
+    form.start_year_2025_or_later? && is_bedsit?
+  end
```
Comment (RachaelBooth) [inline app/models/sales_log.rb:401]: Ah, just noticed - I'm expected some code that actively sets beds to 1 if this is true (probably in derived variables, I think that's where it is for lettings?)

Comment (Dinssa) [reply app/models/sales_log.rb:401]: Ah yes, that's the missing link, good spot. I'll add that in.

**Review by RachaelBooth — APPROVED** [review]

**Pull request merged**